Allow long zone

chrishalcrow opened the pull request:
A suggestion:

Change the layout possibilities from:

```
+-----------------+-----------------+
| [zone1   zone2] | [zone3   zone4] |
+-----------------+-----------------+
| [zone5   zone6] | [zone7   zone8] |
+-----------------+-----------------+
```

to

```
+-----------------+-----------------+
| [zone1   zone2] | [zone3 | [zone4 |
+-----------------+        |        +
| [zone5   zone6] | zone7] | zone8] |
+-----------------+-----------------+
```

Mostly, this allows for a vertical waveforms view without having to use up four zones. So you're able to get layouts like this:

<img width="1470" height="956" alt="Screenshot 2025-07-24 at 12 09 57" src="https://github.com/user-attachments/assets/ec931c04-c6d0-4570-b53b-5b845c07fb7f" />

If you agree, I'll implement for web too!

alejoe91 commented:
Thanks @chrishalcrow 

I wanted to do this, but never found the time. It looks great! Please go ahead and implement for panel too :)

chrishalcrow commented:
Added to web - rewrote the whole web thing as it wasn't really generalising. Hoping the logic is simpler and stupid-er now.

Also added a lil section to the README, to show how to add custom layouts.
